code/datums/extensions/health/atom.dm

```diff
@@ -0,0 +1,397 @@
+ * Override only for modifying damage. Do not override to apply additional effects - Use `post_damage_health()` instead.
+ */
+/atom/proc/damage_health(damage, damage_type = null)
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+	if (using_simple_health())
+		return simple_mod_health(-damage)
+	return health_handler.mod_health(-damage, damage_type)
+
+
+/**
+ * Proc called after health has been decreased.
+ * Provided for override use.
+ */
+/atom/proc/post_damage_health(damage, damage_type = null)
+	return
+
+
+/**
+ * Proc called after any health changes.
+ */
+/atom/proc/post_health_change(damage, damage_type = null)
+	return
+
+
+/**
+ * Immediately sets health to `0` then updates `death_state`, bypassing all other checks and processes.
+ */
+/atom/proc/kill_health()
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+	if (using_simple_health())
+		return simple_mod_health(-simple_max_health)
+	return health_handler.kill()
+
+
+/atom/proc/revive_health()
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+	if (using_simple_health())
+		return simple_mod_health(simple_max_health)
+	return health_handler.revive()
+
+
+/**
+ * Proc called when `death_state` changes.
+ * Provided for override use.
+ */
+/atom/proc/handle_death_change(new_death_state)
+	return
+
+
+/**
+ * Handler for the health extension's `set_max_health()` proc
+ */
+/atom/proc/set_max_health(new_max_health)
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+	if (using_simple_health())
+		simple_max_health = new_max_health
+		return FALSE
+	return health_handler.set_max_health(new_max_health)
+
+
+/**
+ * Handler for the health extension's `set_resistance()` proc
+ */
+/atom/proc/set_damage_resistance(damage_type, resistance_value)
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health() || using_simple_health())
+		return
+	health_handler.set_resistance(damage_type, resistance_value)
+
+
+/**
+ * Handler for the health extension's `set_resistance()` proc
+ */
+/atom/proc/remove_damage_resistance(damage_type)
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health() || using_simple_health())
+		return
+	health_handler.remove_resistance(damage_type)
+
+
+/**
+ * Handler for the health extension's `get_resistance()` proc
+ */
+/atom/proc/get_damage_resistance(damage_type)
+	SHOULD_CALL_PARENT(TRUE)
+	if (!has_health())
+		return
+	if (using_simple_health())
+		return 1
+	return health_handler.get_resistance(damage_type)
+
+
+/**
+ * Handles sending damage state to users on `examine()`.
+ * Overrideable to allow for different messages, or restricting when the messages can or cannot appear.
+ */
+/atom/proc/examine_damage_state(mob/user)
+	if (!has_health())
+		return
+	if (!is_alive())
+		to_chat(user, SPAN_DANGER("It looks broken."))
+		return
+
+	var/health_percentage = get_damage_percentage()
+	switch (health_percentage)
+		if (1)
+			to_chat(user, SPAN_NOTICE("It looks fully intact."))
```

Comment:
````
var/decl/pronouns/G = get_pronouns()
...
to_chat(user, SPAN_NOTICE("[G.He] looks fully intact."))
````
Might be worth adding a `s` to the pronouns decl so you could do `[G.He] look[G.s]`

Reply:
I'll add this in a separate PR once I'm done going over the reviews on this one.